theano/gpuarray/opt.py

```diff
@@ -298,9 +300,7 @@ def apply(self, fgraph):
         target = infer_context_name(*fgraph.inputs)
         for i in fgraph.inputs:
             # Do not move *int* scalar to the GPU.
-            if (isinstance(i.type, tensor.TensorType) and
-                    (i.ndim > 0 or 'int' not in i.dtype) and
-                    "complex" not in i.dtype):
+            if isinstance(i.type, tensor.TensorType) and move_to_gpu(i):
```

Comment:
I remember this was one thing that I want to do it in Graph2Gpu PR. Looks much cleaner. 👍